### PR TITLE
mail-server: Submission of an email server configuration

### PR DESCRIPTION
**Pull request by essandess**
mail-server: Submission of an email server configuration

* postfix for the MTA
* dovecot for the MDA
* solr for fast search
* rspamd for a milter
* clamav for email virus scans
* surrogate TLS certificate and DKIM configuration

#### Description

This email server is a combination of the now-defunct macOS Server.app Mail service with a lot of newer capabilities.

###### Type(s)

- [x] submission
- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6 18G84
Xcode 10.3 10G8

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: mail/mail-server/Portfile [hunk 1]
```diff
@@ -0,0 +1,877 @@
...
+long_description ${description} \
+    Mail server working configuration that provides a basic, working, \
+    easily modifiable mail server. The configuration is built using \
+    postfix for the MTA, dovecot for the MDA, solr for fast search, \
+    rspamd for a milter, and clamav for email virus scans. The \
+    configuration includes a surrogate TLS certificate and DKIM.
+
+homepage                https://www.postfix.org/ \
+                        https://www.dovecot.org/ \
+                        https://rspamd.com/
```
Comment (mf2k) [inline mail/mail-server/Portfile]: It does not really make sense to have more than one `homepage`. The whole point, AFAIK, is a landing place for the command:

`port gohome mailserver`

Please only use one. I'm not sure what it does in this case.

File: mail/mail-server/Portfile [hunk 2]
```diff
@@ -0,0 +1,877 @@
...
+set dovecot2_required_variants \
+                        [lsort {gssapi solr}]
+
+depends_lib-append      port:postfix \
+                        port:dovecot2 \
+                        port:dovecot2-sieve \
+                        port:rspamd \
+                        port:dcc \
+                        port:openssl \
+                        port:sf-pwgen
```
Comment (mf2k) [inline mail/mail-server/Portfile]: Please sort the dependencies by name.

File: mail/mail-server/Portfile [hunk 3]
```diff
@@ -0,0 +1,877 @@
...
+# these dependencies are from variants that may not be active
+depends_lib-append      port:pcre \
+                        port:apache-solr8 \
+                        port:clucene \
+                        port:curl \
+                        port:expat
+
+depends_run-append      port:clamav-server \
+                        port:dns-server \
+                        port:logrotate
```
Comment (mf2k) [inline mail/mail-server/Portfile]: Please sort the `depends_lib` and `depends_run` dependencies too.

File: mail/mail-server/Portfile (hunk 1 above)
Comment (essandess) [reply mail/mail-server/Portfile]: Thanks. I just chose postfix. Done.

File: mail/mail-server/Portfile (hunk 2 above)
Comment (essandess) [reply mail/mail-server/Portfile]: Thanks. Done.

File: mail/mail-server/Portfile (hunk 3 above)
Comment (essandess) [reply mail/mail-server/Portfile]: Also done. Thanks.

**Pull request merged**